sourcetable.go

```diff
@@ -123,3 +127,163 @@ func (m MountEntry) String() string {
 		m.CountryCode, m.Latitude, m.Longitude, nmea, solution, m.Generator, m.Compression,
 		m.Authentication, fee, m.Bitrate, m.Misc)
 }
+
+// DecodeSourcetable parses a sourcetable from an ioreader into a ntrip style source table.
+func DecodeSourcetable(in []byte) (Sourcetable, []error) {
+	table := Sourcetable{}
+	str := string(in[:])
+	var errs []error
+
+	lines := strings.Split(str, "\n")
+
+	for lineNo, rawLine := range lines {
+		line := strings.TrimSpace(rawLine)
+
+		if line == "" {
+			continue
+		}
+
+		if line == "ENDSOURCETABLE" {
+			break
+		}
+
+		switch line[:3] {
+		case "CAS":
+			caster, err := buildCasterEntry(line)
+			if err != nil {
+				errs = append(errs, errors.Wrapf(err, "parsing caster %v", lineNo))
+			}
+			table.Casters = append(table.Casters, caster)
+		case "NET":
+			net, err := buildNetEntry(line)
+			if err != nil {
+				errs = append(errs, errors.Wrapf(err, "parsing line %v", lineNo))
+			}
+			table.Networks = append(table.Networks, net)
+		case "STR":
+			mount, err := buildMountEntry(line)
+			if err != nil {
+				errs = append(errs, errors.Wrapf(err, "parsing line %v", lineNo))
+			}
+			table.Mounts = append(table.Mounts, mount)
+		}
+
+	}
+
+	return table, errs
+}
+
+func buildCasterEntry(casterString string) (CasterEntry, error) {
```

Comment:
Again, I think Parse might be a better word than build here. ParseSourcetable calls ParseCasterEntry, etc.